src/filter/http_context.rs

```diff
@@ -164,8 +164,7 @@ impl Filter {
                                 self.context_id, selector_item.selector
                             );
                             match &selector_item.default {
-                                // skipping the entire descriptor
-                                None => "".to_string(),
+                                None => return None, // skipping the entire descriptor
```

Comment:
I think returning an empty string was [intentional](https://github.com/Kuadrant/wasm-shim/commit/59033b0b4f89290391c8eee9acf00fde51fe6654), as part of avoiding the unsafe default of skipping the limit when a descriptor expected to exist is missing. Rather, it seemed safer to occasionally over-limit (i.e. by expanding the scope of a counter to a common, "catch-all", counter when a selector is evaluated to `""`, because the descriptor is missing), than to under-limit (by not enforcing any limit at all).

Either way, the user will have to debug and figure out why it's not rate limiting as expected.

Reply:
For selectors in conditions, it makes sense to return something when the property is not found in order to apply operators. However, for `data.selectors`, there is the `default` attribute to cover the use case where the property does not exist https://github.com/Kuadrant/wasm-shim/blob/main/src/configuration.rs#L7-L20

```rust
pub struct SelectorItem {
    // Selector of an attribute from the contextual properties provided by kuadrant
    // during request and connection processing
    pub selector: String,

    // If not set it defaults to `selector` field value as the descriptor key.
    #[serde(default)]
    pub key: Option<String>,

    // An optional value to use if the selector is not found in the context.
    // If not set and the selector is not found in the context, then no data is generated.
    #[serde(default)]
    pub default: Option<String>,
}
```

Check the doc string of the `default` field. 

IMO it makes sense to assume that if the property does not exist AND there is no default, no descriptor entry is generated. If you want to generate descriptor entry no matter the property exists or not, set `default`. 

At the kuadrant control plane level, when there is RLP CR counter, the `selector.data[].selector` item will be generated. If we want to reach limitador no matter the property exists or not, the controller should add a default. But that's control plane level call. IMO, the tool (the wasm shim) should cover all use cases.

Reply:
Gotcha. Thanks for explaining.